custom-routes-vis/src/main/webapp/script.js

```diff
@@ -73,6 +77,40 @@ function updateCoordinates(lat, lng, containerId) {
       "Latitude: " + lat + "<br>Longitude: " + lng + "<br>";
 }
 
+/**
+ * Updates the deep linking url to various test apps on the page.
+ * TODO: Update Route Token if we get access to Routes Preferred API.
+ */
+function updateDeepLinkingUrl() {
+  if (document.getElementById('origin-coordinates').innerHTML != '' && 
+      document.getElementById('destination-coordinates').innerHTML != '') {
+    document.getElementById(IOS_URL_ID).innerHTML =
+        '<a href=navsdkdemo://advanced?originLat=' + originDestinationMarkers[0].position.lat()
+        + '&originLng=' + originDestinationMarkers[0].position.lng() + '&destLat=' 
+        + originDestinationMarkers[1].position.lat() + '&destLng=' 
+        + originDestinationMarkers[1].position.lng() + '&routeToken=TOKEN>' 
+        + 'IOS Test App' + '</a>';
+    document.getElementById(V1_ANDROID_URL_ID).innerHTML =
+        '<a href=navsdk://fragmentactivity?originlat=' + originDestinationMarkers[0].position.lat() 
+        + '&originlng=' + originDestinationMarkers[0].position.lng() + '&destinationlat=' 
+        + originDestinationMarkers[1].position.lat() + '&destinationlng='
+        + originDestinationMarkers[1].position.lng() + '&precomputedroutetoken=TOKEN>' 
+        + 'Android V1 Test App' + '</a>';
+    document.getElementById(V2_ANDROID_URL_ID).innerHTML =
+        '<a href=navsdk://supportnavmapfragmentactivity?originlat=' 
+        + originDestinationMarkers[0].position.lat() + '&originlng='
+        + originDestinationMarkers[0].position.lng() + '&destinationlat=' 
+        + originDestinationMarkers[1].position.lat() + '&destinationlng='
+        + originDestinationMarkers[1].position.lng() + '&precomputedroutetoken=TOKEN>' 
+        + 'Android V2 Test App' + '</a>';
+  } else if (document.getElementById('origin-coordinates').innerHTML == '' || 
+      document.getElementById('destination-coordinates').innerHTML == '') { 
+    document.getElementById(IOS_URL_ID).innerHTML = '';
+    document.getElementById(V1_ANDROID_URL_ID).innerHTML = '';
+    document.getElementById(V2_ANDROID_URL_ID).innerHTML = '';
```

Comment:
This looks really repetitive . Please simplify it.

Reply:
The current solution I can think of is either making a javaScript Object with the three container ID's, or an array and then having a for loop to make all the innerHTML empty, I'll continue thinking of other ways, but do others have ideas? @lawrencezjh @ChenyuHeidiZhang @carmenBen ?